src/Core/Houston.vala

```diff
@@ -17,85 +17,154 @@
 */
 
 public class AppCenterCore.Houston : Object {
+    public signal void refreshed ();
+
     private const string HOUSTON_API_URL = "https://developer.elementary.io/api";
+    private const string NEWEST_CATEGORY = "Newest";
+    private const string UPDATED_CATEGORY = "Updated";
+    private const string APPS_KEY = "apps";
 
     private Soup.Session session;
+    private GLib.KeyFile cache_file = null;
```

Comment:
Ah, I haven't noticed that it is inded a singleton, well then, that's not required.